### PR TITLE
Rename kuzzle-reloader to ergol, extract reloader functions, add github action & workflows

### PR DESCRIPTION
**Pull request by MathieuVeber**
# What does this PR do?

- Rename kuzzle-reloader to ergol
- Extract reloader functions
- Add proper cli options with flags
- Add config file handling
- Adds a linter and ensure that there is no lint errors for each PR.
- Also automates the release on npm when pushing on master.

## Workflows steps

- Pull request: lint test
- Push on master: lint test and (if successful) deployment on NPM

### CODE REVIEW

File: .github/actions/lint/action.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: ESLint
+description: Run ESLint
+runs:
+  using: "composite"
+  steps:
+    - name: Install deps
```
Comment (rolljee) [inline .github/actions/lint/action.yml]: Name here does not truly reflect what's happening In there

File: .github/workflows/push_master.workflow.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1.4.4
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/lint
+
+  npm-deploy:
+    name: Publish Package to NPM.js
+    runs-on: ubuntu-18.04
+    needs: [lint]
```
Comment (rolljee) [inline .github/workflows/push_master.workflow.yml:35]: Add timeout here even if it's a public repository

File: .github/workflows/push_master.workflow.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-18.04
+    steps:
```
Comment (rolljee) [inline .github/workflows/push_master.workflow.yml:13]: Add timeout here even if it's a public repository

File: .github/workflows/pull_request.workflow.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+name: Pull request checks
+
+on: [pull_request]
+
+jobs:
+  lint:
+    name: Lint
```
Comment (rolljee) [inline .github/workflows/pull_request.workflow.yml:7]: Add timeout here even if it's a public repository

**Review by rolljee — CHANGES_REQUESTED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Review by rolljee — APPROVED** [review]

**Pull request merged**